src/components/DeleteProjectButton/DeleteProjectButton.js

```diff
@@ -0,0 +1,131 @@
+import PropTypes from 'prop-types'
+import React from 'react'
+import { Link } from 'react-router-dom'
+import { ButtonCaution, ButtonSecondary } from '../generic/buttons'
+import Modal, { RightFooter } from '../generic/Modal/Modal'
+import {
+  DeleteProjectButtonCautionWrapper,
+  WarningText,
+} from '../pages/collectRecordFormPages/CollectingFormPage.Styles'
+import LoadingModal from '../LoadingModal/LoadingModal'
+import useCurrentProjectPath from '../../library/useCurrentProjectPath'
+import language from '../../language'
+import { IconDeleteForever } from '../icons'
+
+const DeleteProjectButton = ({
+  currentPage,
+  errorData,
+  isLoading,
+  hasSampleUnits,
+  hasOtherUsers,
+  isOpen,
+  modalText,
+  deleteProject,
+  onDismiss,
+  openModal,
+}) => {
+  const currentProjectPath = useCurrentProjectPath()
+
+  const mainContentPageTwo = (
+    <>
+      <p>{modalText.confirmDeleteText1}</p>
+      <ul>
+        {errorData.map((error) => (
+          <li key={error.id}>
+            <Link to={`${currentProjectPath}/submitted/${error.protocol}/${error.id}`}>
+              {error.sampleUnitLabel}
+            </Link>
+          </li>
+        ))}
+      </ul>
+      <p>{modalText.confirmDeleteText2}</p>
+    </>
+  )
+
+  const footerContentPageOne = (
+    <RightFooter>
+      <ButtonSecondary onClick={onDismiss}>{modalText.no}</ButtonSecondary>
+      <ButtonCaution disabled={isLoading} onClick={deleteProject}>
+        {modalText.yes}
+      </ButtonCaution>
+    </RightFooter>
+  )
+
+  const footerContentPageTwo = (
+    <RightFooter>
+      <ButtonSecondary onClick={onDismiss}>Close</ButtonSecondary>
+    </RightFooter>
+  )
+
+  const mainContent = (
+    <>
+      {currentPage === 1 && modalText.prompt}
```

Comment:
I dont think the use of `&&` here creates a logical issue, however I think its best to use a ternary to be explicit about what is expected in the falsey case (we want null). I just use ternary on autopilot so I dont have to think about  if && will create a bug. 

https://kentcdodds.com/blog/use-ternaries-rather-than-and-and-in-jsx

Reply:
good catch ! will update to ternary